src/Sylius/Bundle/CoreBundle/Checkout/CheckoutResolver.php

```diff
@@ -31,19 +29,8 @@ public function __construct(
         private CartContextInterface $cartContext,
         private CheckoutStateUrlGeneratorInterface $urlGenerator,
         private RequestMatcherInterface $requestMatcher,
-        private FactoryInterface|StateMachineInterface $stateMachineFactory,
+        private StateMachineInterface $stateMachineFactory,
```

Comment:
```suggestion
        private StateMachineInterface $stateMachine,
```
and the same in other places

Reply:
https://github.com/Sylius/Sylius/pull/17002